test/performance/search/index_construction_benchmark.cpp

```diff
@@ -9,218 +9,188 @@
 
 #include <seqan3/alphabet/aminoacid/aa27.hpp>
 #include <seqan3/alphabet/nucleotide/dna4.hpp>
+#include <seqan3/core/type_traits/lazy.hpp>
 #include <seqan3/range/views/persist.hpp>
 #include <seqan3/range/views/rank_to.hpp>
-#include <seqan3/range/views/to.hpp>
-#include <seqan3/range/views/to_rank.hpp>
 #include <seqan3/search/fm_index/all.hpp>
 #include <seqan3/test/performance/sequence_generator.hpp>
 #include <seqan3/test/seqan2.hpp>
 
+
 #if SEQAN3_HAS_SEQAN2
 #include <seqan/index.h>
 #endif
 
-static void arguments(benchmark::internal::Benchmark* b)
+static constexpr int32_t max_length{50'000};
+static constexpr size_t seed{0x6126f};
+
+static void arguments(benchmark::internal::Benchmark * b)
 {
     for (int32_t length : {50, 5000, 50'000})
+    {
+        if (length > max_length)
+            throw std::logic_error{"Increase max_length to at least " + std::to_string(length)};
+
         b->Args({length, 5});
+    }
 
     b->Args({500, 1'000});
 }
 
-struct fm_index_seqan2;
-struct fm_index_seqan3;
-struct bi_fm_index_seqan2;
-struct bi_fm_index_seqan3;
+enum class tag
+{
+    fm_index,
+    bi_fm_index
+};
 
-template <typename index_tag_t, typename rng_t>
-class sequence_generator
+struct sequence_store_seqan3
 {
-private:
-    size_t length{50'000};
-    size_t collection_size{5};
+    std::vector<seqan3::dna4> const dna4_rng{seqan3::test::generate_sequence<seqan3::dna4>(max_length, 0, seed)};
+    std::vector<seqan3::aa27> const aa27_rng{seqan3::test::generate_sequence<seqan3::aa27>(max_length, 0, seed)};
+    std::string const char_rng{seqan3::test::generate_numeric_sequence<uint8_t>(max_length, 0, 253, seed)
+                               | seqan3::views::persist
+                               | seqan3::views::rank_to<char>
+                               | seqan3::views::to<std::string>};
+};
+
+sequence_store_seqan3 store3;
```

Comment:
```suggestion
sequence_store_seqan3 store{};
```
I like the design :)